From 157ce65588ec8ef4ff8f805f7b8a64fefdd9edec Mon Sep 17 00:00:00 2001
From: Stephan Bergmann <sbergman@redhat.com>
Date: Fri, 08 Mar 2013 11:07:11 +0000
Subject: Work around GCC 4.8 -Werror=virtual-move-assign

...complaining that "defaulted move assignment for ‘cppcanvas::SpriteCanvas’
calls a non-trivial move assignment operator for virtual base
‘cppcanvas::BitmapCanvas’."

Change-Id: I4dc92176dbec62f0ae4fa7d2842073077ecb8fc3
---
diff --git a/cppcanvas/inc/cppcanvas/spritecanvas.hxx b/cppcanvas/inc/cppcanvas/spritecanvas.hxx
index 59aa3da..6a606b3 100644
--- a/cppcanvas/inc/cppcanvas/spritecanvas.hxx
+++ b/cppcanvas/inc/cppcanvas/spritecanvas.hxx
@@ -24,6 +24,7 @@
 #include <osl/diagnose.h>
 #include <basegfx/vector/b2dsize.hxx>
 
+#include <boost/noncopyable.hpp>
 #include <boost/shared_ptr.hpp>
 
 
@@ -48,7 +49,7 @@ namespace cppcanvas
 
     /** SpriteCanvas interface
      */
-    class SpriteCanvas : public virtual BitmapCanvas
+    class SpriteCanvas : public virtual BitmapCanvas, private boost::noncopyable
     {
     public:
         virtual bool                    updateScreen( bool bUpdateAll ) const = 0;
--
cgit v0.9.0.2-2-gbebe
